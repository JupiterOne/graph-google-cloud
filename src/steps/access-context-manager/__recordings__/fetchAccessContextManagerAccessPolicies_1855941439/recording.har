{
  "log": {
    "_recordingName": "fetchAccessContextManagerAccessPolicies",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 739,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/x-www-form-urlencoded"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "739"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 300,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/x-www-form-urlencoded",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 572,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 572,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Wed, 02 Jun 2021 14:22:23 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 511,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-06-02T14:22:23.123Z",
        "time": 212,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 212
        }
      },
      {
        "_id": "e7ae7a9d3285d95d4330a7c239aa61b7",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/5.0.2 gl-node/14.15.4 auth/7.1.0"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/5.0.2 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "accesscontextmanager.googleapis.com"
            }
          ],
          "headersSize": 581,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "parent",
              "value": "organizations/958457776463"
            }
          ],
          "url": "https://accesscontextmanager.googleapis.com/v1/accessPolicies?parent=organizations%2F958457776463"
        },
        "response": {
          "bodySize": 369,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 369,
            "text": "{\"accessPolicies\":[{\"name\":\"accessPolicies/368605946207\",\"parent\":\"organizations/958457776463\",\"title\":\"default policy\",\"etag\":\"64186ca0c82b01b4\"}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Wed, 02 Jun 2021 14:22:24 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 488,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-06-02T14:22:23.374Z",
        "time": 829,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 829
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
