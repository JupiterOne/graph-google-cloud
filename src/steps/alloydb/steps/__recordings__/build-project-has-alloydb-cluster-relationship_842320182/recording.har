{
  "log": {
    "_recordingName": "build-project-has-alloydb-cluster-relationship",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "6.0.5"
    },
    "entries": [
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 709,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/x-www-form-urlencoded"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/9.4.1"
            },
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gl-node/18.18.2"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "709"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 293,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/x-www-form-urlencoded",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 1160,
          "content": {
            "encoding": "utf-8",
            "mimeType": "application/json; charset=UTF-8",
            "size": 1160,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Tue, 25 Jun 2024 12:21:32 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 390,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2024-06-25T12:21:31.870Z",
        "time": 352,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 352
        }
      },
      {
        "_id": "23a717153d5a0e7db6eb50c807895f67",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/7.0.1 gl-node/18.18.2"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/7.0.1 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "name": "host",
              "value": "cloudresourcemanager.googleapis.com"
            }
          ],
          "headersSize": 1322,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://cloudresourcemanager.googleapis.com/v3/projects/j1-gc-integration-dev-v3"
        },
        "response": {
          "bodySize": 397,
          "content": {
            "encoding": "utf-8",
            "mimeType": "application/json; charset=UTF-8",
            "size": 397,
            "text": "{\"name\":\"projects/167984947943\",\"parent\":\"folders/306329820365\",\"projectId\":\"j1-gc-integration-dev-v3\",\"state\":\"ACTIVE\",\"displayName\":\"j1-gc-integration-dev-v3\",\"createTime\":\"2021-05-31T15:15:56.125Z\",\"updateTime\":\"2024-05-15T11:48:40.174923Z\",\"etag\":\"W/\\\"8f56735f6f3d890d\\\"\",\"labels\":{\"firebase\":\"enabled\"}}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Tue, 25 Jun 2024 12:21:33 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 367,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2024-06-25T12:21:32.230Z",
        "time": 1539,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 1539
        }
      },
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 1,
        "cache": {},
        "request": {
          "bodySize": 709,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/x-www-form-urlencoded"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/9.4.1"
            },
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gl-node/18.18.2"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "709"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 293,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/x-www-form-urlencoded",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 1195,
          "content": {
            "encoding": "utf-8",
            "mimeType": "application/json; charset=UTF-8",
            "size": 1195,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Tue, 25 Jun 2024 12:21:33 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 390,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2024-06-25T12:21:33.793Z",
        "time": 174,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 174
        }
      },
      {
        "_id": "faaa92f3d61304671425d51a081e923f",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/7.0.1 gl-node/18.18.2"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/7.0.1 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "name": "host",
              "value": "alloydb.googleapis.com"
            }
          ],
          "headersSize": 1317,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://alloydb.googleapis.com/v1/projects/j1-gc-integration-dev-v3/locations/-/clusters"
        },
        "response": {
          "bodySize": 981,
          "content": {
            "encoding": "utf-8",
            "mimeType": "application/json; charset=UTF-8",
            "size": 981,
            "text": "{\"clusters\":[{\"name\":\"projects/j1-gc-integration-dev-v3/locations/us-east4/clusters/test-cluster-1\",\"uid\":\"7bdaf17a-f3ec-480b-affa-4adbfae59791\",\"createTime\":\"2024-04-18T01:57:32.802698772Z\",\"updateTime\":\"2024-06-25T12:02:19.407224129Z\",\"state\":\"READY\",\"databaseVersion\":\"POSTGRES_15\",\"network\":\"projects/167984947943/global/networks/test-network\",\"reconciling\":false,\"automatedBackupPolicy\":{\"enabled\":false,\"weeklySchedule\":{\"startTimes\":[{\"hours\":8}],\"daysOfWeek\":[\"MONDAY\",\"TUESDAY\",\"WEDNESDAY\",\"THURSDAY\",\"FRIDAY\",\"SATURDAY\",\"SUNDAY\"]},\"backupWindow\":\"3600s\",\"timeBasedRetention\":{\"retentionPeriod\":\"1209600s\"},\"location\":\"us-east4\"},\"encryptionInfo\":{\"encryptionType\":\"GOOGLE_DEFAULT_ENCRYPTION\"},\"clusterType\":\"PRIMARY\",\"continuousBackupConfig\":{\"enabled\":true,\"recoveryWindowDays\":14},\"continuousBackupInfo\":{\"encryptionInfo\":{\"encryptionType\":\"GOOGLE_DEFAULT_ENCRYPTION\"},\"enabledTime\":\"2024-04-18T01:57:32.168373933Z\",\"schedule\":[\"MONDAY\",\"TUESDAY\",\"WEDNESDAY\",\"THURSDAY\",\"FRIDAY\",\"SATURDAY\",\"SUNDAY\"]},\"networkConfig\":{\"network\":\"projects/167984947943/global/networks/test-network\"}}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Tue, 25 Jun 2024 12:21:35 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 367,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2024-06-25T12:21:33.984Z",
        "time": 2027,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 2027
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
