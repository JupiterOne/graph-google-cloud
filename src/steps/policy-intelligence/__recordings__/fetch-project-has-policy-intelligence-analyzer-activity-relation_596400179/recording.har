{
  "log": {
    "_recordingName": "fetch-project-has-policy-intelligence-analyzer-activity-relation",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "6.0.5"
    },
    "entries": [
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 709,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/x-www-form-urlencoded"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/9.4.1"
            },
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gl-node/18.18.2"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "709"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 293,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/x-www-form-urlencoded",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 1280,
          "content": {
            "encoding": "utf-8",
            "mimeType": "application/json; charset=UTF-8",
            "size": 1280,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Tue, 11 Jun 2024 12:45:20 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 390,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2024-06-11T12:45:20.666Z",
        "time": 297,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 297
        }
      },
      {
        "_id": "913f50c7f8ec0c0ac8a2cbb4d8141b7e",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/7.0.1 gl-node/18.18.2"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/7.0.1 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "name": "host",
              "value": "cloudresourcemanager.googleapis.com"
            }
          ],
          "headersSize": 308,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://cloudresourcemanager.googleapis.com/v3/projects/j1-gc-integration-dev-v2"
        },
        "response": {
          "bodySize": 664,
          "content": {
            "encoding": "utf-8",
            "mimeType": "application/json; charset=UTF-8",
            "size": 664,
            "text": "{\"error\":{\"code\":401,\"message\":\"Request had invalid authentication credentials. Expected OAuth 2 access token, login cookie or other valid authentication credential. See https://developers.google.com/identity/sign-in/web/devconsole-project.\",\"errors\":[{\"message\":\"Invalid Credentials\",\"domain\":\"global\",\"reason\":\"authError\",\"location\":\"Authorization\",\"locationType\":\"header\"}],\"status\":\"UNAUTHENTICATED\",\"details\":[{\"@type\":\"type.googleapis.com/google.rpc.ErrorInfo\",\"reason\":\"ACCESS_TOKEN_TYPE_UNSUPPORTED\",\"metadata\":{\"method\":\"google.cloud.resourcemanager.v3.Projects.GetProject\",\"service\":\"cloudresourcemanager.googleapis.com\"}}]}}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "www-authenticate",
              "value": "Bearer realm=\"https://accounts.google.com/\", error=\"invalid_token\""
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "date",
              "value": "Tue, 11 Jun 2024 13:10:46 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 453,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 401,
          "statusText": "Unauthorized"
        },
        "startedDateTime": "2024-06-11T13:10:44.413Z",
        "time": 1766,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 1766
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
