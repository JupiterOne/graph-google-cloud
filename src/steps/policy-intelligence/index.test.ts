import {
  createMockStepExecutionContext,
  executeStepWithDependencies,
  Recording,
  StepTestConfig,
} from '@jupiterone/integration-sdk-testing';
import {
  fetchPolicyAnalyzer,
} from '.';
import { integrationConfig } from '../../../test/config';
import { setupGoogleCloudRecording, getMatchRequestsBy } from '../../../test/recording';
import { IntegrationConfig } from '../../types';
import {
  POLICY_INTELLIGENCE_ANALYZER_TYPE,
  STEP_POLICY_INTELLIGENCE_ANALYZER_ACTIVITY,
} from './constants';
import { invocationConfig } from '../..';

const tempNewAccountConfig = {
  ...integrationConfig,
  serviceAccountKeyFile: integrationConfig.serviceAccountKeyFile.replace(
    'j1-gc-integration-dev-v2',
    'j1-gc-integration-dev-v3',
  ),
  serviceAccountKeyConfig: {
    ...integrationConfig.serviceAccountKeyConfig,
    project_id: 'j1-gc-integration-dev-v3',
  },
};

describe('#fetchPolicyAnalyzer', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupGoogleCloudRecording({
      directory: __dirname,
      name: 'fetchPolicyAnalyzer',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: tempNewAccountConfig,
    });

    await fetchPolicyAnalyzer(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e.type === POLICY_INTELLIGENCE_ANALYZER_TYPE,
      ),
    ).toMatchGraphObjectSchema({
      _class: ['Service'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_cloud_policy_intelligence_analyzer' },
          name: { type: 'string' },
          category: { type: 'array' },
          function: { type: 'array' },
        },
      },
    });
  });
});

describe(`#${STEP_POLICY_INTELLIGENCE_ANALYZER_ACTIVITY}`, () => {
  let recording: Recording;
  afterEach(async () => {
    if (recording) await recording.stop();
  });
  test(STEP_POLICY_INTELLIGENCE_ANALYZER_ACTIVITY, async () => {
    recording = setupGoogleCloudRecording({
      name: STEP_POLICY_INTELLIGENCE_ANALYZER_ACTIVITY,
      directory: __dirname,
      options: {
        matchRequestsBy: getMatchRequestsBy(integrationConfig),
        recordFailedRequests: true,
      },
    });
    const stepTestConfig: StepTestConfig = {
      stepId: STEP_POLICY_INTELLIGENCE_ANALYZER_ACTIVITY,
      instanceConfig: integrationConfig,
      invocationConfig: invocationConfig as any,
    };
    const result = await executeStepWithDependencies(stepTestConfig);
    expect(result).toMatchStepMetadata(stepTestConfig);
  }, 300000);
});