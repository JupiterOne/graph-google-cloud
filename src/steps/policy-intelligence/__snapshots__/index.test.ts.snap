// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchPolicyAnalyzer should collect data 1`] = `
{
  "collectedEntities": [],
  "collectedRelationships": [],
  "encounteredTypes": [],
  "numCollectedEntities": 0,
  "numCollectedRelationships": 0,
}
`;
